watershed_workflow/condition.py

```diff
@@ -392,5 +391,185 @@ def fill_gaps(img_in, nodata=np.nan):
+        this will enforce bank node is at a higher elevation than the stream bed elevation 
+    depress_by: int, optional
+        if the depression is not captured well in DEM, the river-mesh elements (streambed) is lowered by this number
+
+    Returns
+    -------
+    m2: watershed_workflow.mesh.Mesh2D object
+        a 2D mesh with conditioned stream network           
+    """
+    river_corr_ids=[] # collecting IDs of all nodes in the river/stream
+    for node in river.preOrder():
+        for elem in node.elements:
+            for id in elem:
+                if id not in river_corr_ids: 
+                    river_corr_ids.append(id)
+   
+    # conditoning of stream-bed profiles to enforce typical channel depths, large-scale 
+    # topographic gradients in the streambeds, and connectivity through culverts that pass under road and railway embankments
+    if smooth:        
+        for node in river.preOrder():  # reachwise smoothing 
+            smooth_profile(node, use_parent=use_parent, lower=lower) # adds smooth profile in node properties
+
+        network_sweep(river, depress_by=depress_by, use_nhd_elev=use_nhd_elev) # network-wide conditioning 
+
+    # transferring network-scale-conditioned stream-bed elevations onto the mesh     
+    for node in river.preOrder():
+        
+        if smooth:
+            profile=node.properties["SmoothProfile"]
+        else:
+            profile=get_profile(node) # if only centerline elevation is to be use, without any conditioning
+
+        for i, elem in enumerate(node.elements): 
+            
+            if i ==0: # for the first point
+                m2.coords[elem[0]][2] = m2.coords[elem[-1]][2] = profile[i,1]
+                
+            # assgining elevations to the upstream points of each river-corridor element (quads)
+            for coord_id in elem[1:-1]:
+                m2.coords[coord_id][2] = profile[i+1,1]
+
+            # this to ensure that a diked channel passing over/around a pond or reservoirs
+            #  do not have bank-nodes fall into the depression  
+            if treat_banks: 
+                bank_node_ids=bank_nodes_from_elem(elem, m2)
+                for node_id in bank_node_ids:
+                    if node_id not in river_corr_ids:
+                        if m2.coords[node_id][2] < min(profile[i+1,1], profile[i,1]):
+                                logging.info(f"raised node {node_id} for bank integrity") 
+                                m2.coords[node_id][2]= 0.5*(profile[i,1]+profile[i+1,1])+0.55 
+
+
+def get_profile(node):
+    """for a given node, genereates a bedprofile using elevations on the node.segment"""
+    stream_bed_coords=list(reversed(node.segment.coords)) # node that node_elems are downstream to upstream, while segment coords are upstream to downstream
+    dists=[math.dist(stream_bed_coords[0],point) for point in stream_bed_coords]
```

Comment:
I think this is new in python 3.8.

That's probably OK -- I'm not sure what we have the min python version set at.  Note to check and make sure it is >=3.8

Reply:
ok, it's 3.10